c2corg_api/ext/colander_ext.py

```diff
@@ -53,11 +53,16 @@ def deserialize(self, node, cstruct):
         if cstruct is null or cstruct == '':
             return null
         try:
-            geojson = json.loads(cstruct)
-            geom_type = geojson['type'].upper()
+            geojson_data = json.loads(cstruct)
+            geom_type = geojson_data['type'].upper()
             allowed_types = self.geometry_types
             if geom_type in allowed_types:
-                return wkbelement_from_geojson(geojson, self.srid)
+                validation = geojson.is_valid(geojson_data)
```

Comment:
I think this should be:
```
validation = geojson.is_valid(geojson.loads(cstruct))
```

Though it would be a pity that the string is parsed twice. It would also be good to test if 3D/4D geometries still work.